### ISSUE TITLE
TypeScript support

### ISSUE THREAD

**Issue by styfle** [substack/hyperx#45]
This might not be possible but I'm wondering if anyone had success with type checking the ES6 tagged template string. Before hyperx, I wrote something similar called [typed-tmpl](https://github.com/styfle/typed-tmpl) which type checks the variables but check the html such as a missing an angle bracket like the [TSX](http://www.typescriptlang.org/docs/handbook/jsx.html) syntax does.

Has any had success with validating the HTML in the string?

**Comment by jbucaran** [comment]
@styfle Nope, but I'd like to know if you are planning to create a Typex or similar generic module like Hyperx that could be plugged into frameworks, etc.

**Comment by styfle** [comment]
@jbucaran I have not planned anything but I have been thinking about this problem for several years and I just heard about hyperx today.
